### PR TITLE
cron task to clean up the repository

### PR DESCRIPTION
**Pull request by kakj-go**
#### What type of PR is this?
feature

#### What this PR does / why we need it:
If you do not clean up the git warehouse regularly, too long a time will cause the access to the warehouse to time out

### CODE REVIEW

File: modules/gittar/pkg/gc/gc.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	wait.Wait()
+}
+
+// execute the git gc command and print the returned information or error
+func doGcCommand(path string) {
+	// remove spaces to prevent injection attacks
+	path = strings.Replace(path, " ", "", -1)
+	cmd := exec.Command("sh", "-c", "cd '"+path+"' && git gc")
+	logrus.Infof("gc: start gc path: %v", path)
+	stdout, err := cmd.StdoutPipe()
```
Comment (sfwn) [inline modules/gittar/pkg/gc/gc.go:92]: how to handle stderr?

File: modules/gittar/pkg/gc/gc.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	"os/exec"
+	"strings"
+
+	"github.com/erda-project/erda/modules/gittar/pkg/limit_sync_group"
+	"github.com/erda-project/erda/pkg/cron"
+	"github.com/sirupsen/logrus"
+)
+
+// all gittar repository root address
+const repositoryRootAddr = "/repository"
```
Comment (sfwn) [inline modules/gittar/pkg/gc/gc.go:28]: Get repositoryRootAddress from config.
If someone changes the root addr one day, it is easy for him to be unclear about the gc configuration, leading to a mismatch.

File: modules/gittar/pkg/limit_sync_group/group.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package limit_sync_group
+
+import "sync"
+
+// encapsulate sync.WaitGroup
+// realize a controllable number of WaitGroup
+type limitSyncGroup struct {
+	c  chan struct{}
```
Comment (sfwn) [inline modules/gittar/pkg/limit_sync_group/group.go:21]: use buffered channel is enough.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/gittar/pkg/gc/gc.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+// gc repository concurrent number
+const concurrentNum = 1
+
+// gc repository cron time
+const cronExpression = "0 0 1 * * ?"
```
Comment (QvodSoldier) [inline modules/gittar/pkg/gc/gc.go:34]: do we need get cron expresssion from config? Maybe there were some scenes need different gc cycle

**Comment by QvodSoldier** [comment]
I found git gc have a --auto options [git gc](https://git-scm.com/docs/git-gc), but i don't sure is right for us, i think we can use this fix for now.

**Pull request closed without merging**